cards/src/main/java/com/decathlon/vitamin/compose/cards/VitaminCards.kt

```diff
@@ -0,0 +1,258 @@
+     * @param shape to change the shape of the card.
+     */
+    @Composable
+    fun SideImage(
+        painter: Painter,
+        contentDescription: String?,
+        content: (@Composable SideImageContent.() -> Unit),
+        modifier: Modifier = Modifier,
+        actionContent: @Composable (SideImageActionContent.() -> Unit)? = null,
+        shape: Shape = RoundedCornerShape(4.dp),
+    ) = Card(modifier = modifier, shape = shape) {
+        Column {
+            if (actionContent != null) {
+                SideImageActionContent.actionContent()
+            }
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.spacedBy(16.dp),
+                modifier = Modifier
+                    .padding(
+                        top = if (actionContent != null) {
+                            0.dp
+                        } else {
+                            16.dp
+                        },
+                        bottom = 16.dp,
+                        start = 16.dp,
+                        end = 16.dp
+                    )
+            ) {
+                CardImage(
+                    painter = painter,
+                    contentDescription = contentDescription,
+                    isSideImage = true,
+                    modifier = Modifier.weight(weight = 1f).aspectRatio(ratio = 1f)
+                )
+                Box(Modifier.weight(weight = 3f)) {
+                    SideImageContent.content()
+                }
+            }
+        }
+    }
+
+    /**
+     * A card without image
+     * @param content The content of the card. Can be a composable from [WithoutImageContent] or a custom composable.
+     * @param modifier The `Modifier` to be applied to the all card.
+     * @param shape to change the shape of the card.
+     */
+    @Composable
+    fun WithoutImage(
+        content: (@Composable WithoutImageContent.() -> Unit),
+        modifier: Modifier = Modifier,
+        shape: Shape = RoundedCornerShape(4.dp),
+    ) = Card(modifier = modifier, shape = shape) {
+        Box(modifier = Modifier.padding(16.dp)) {
+            WithoutImageContent.content()
+        }
+    }
+}
+
+@Composable
+internal fun CardImage(
+    painter: Painter,
+    contentDescription: String?,
+    modifier: Modifier = Modifier,
+    overlay: @Composable (() -> Unit)? = null,
+    imageOverlayAlignment: Alignment = Alignment.TopStart,
+    withGradient: Boolean = false,
+    isSideImage: Boolean = false,
+) {
+    Box(modifier = modifier) {
+        Image(
+            painter = painter,
+            contentDescription = contentDescription,
+            contentScale = ContentScale.FillWidth,
+            modifier = when {
+                withGradient -> {
+                    Modifier
+                        .fillMaxWidth().withGradient(vtmnLightColorPalette.vtmnBackgroundPrimaryReversed)
```

Comment:
Pretty sure it isn't a good idea to use a color from the palette. Probably an issue in the design specification.